src/libpgagroal/configuration.c

```diff
@@ -2395,6 +2397,29 @@ is_same_server(struct server* s1, struct server* s2)
    }
 }
 
+static bool
+is_same_tls(struct server* src, struct server* dst)
+{
+   if (src->tls != dst->tls)
+   {
+      return false;
+   }
+
+   if (!src->tls)
+   {
+      return true;
+   }
+
+   if (strcmp(src->tls_cert_file, dst->tls_cert_file) ||
```

Comment:
You should reverse this check

Reply:
I'm sorry I wrote unclear code (just for cleaness). I will modify here to improve readability.

But actually strcmp will return 0 only if two strings are same, and every nonzero value will be considered to be true🥺

Could you reconsider if my code is wrong?🥺 Thank you so much and apologize again for its confusion🥺

Reply:
https://pastebin.ubuntu.com/p/Dgd55gBSZW/
I wrote a unittest for this function. The code behavior seems consistent with my expectations. Could you review my unittest to see is any bias on my understanding to this feature? Thank you very much!

Or if you are concerning about line 2410? In this line, I think, if we don't use `tls` at all, there's no need to compare the rest of the config. So we can directly return true.

Additionally I found in our project all of the strcmp is used like mine. So could you give me a little further suggestions?